libvmi/driver/kvm/kvm_events.c

```diff
@@ -0,0 +1,835 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government
+ * retains certain rights in this software.
+ *
+ * Author: Mathieu Tarral (mathieu.tarral@ssi.gouv.fr)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "private.h"
+#include "msr-index.h"
+#include "kvm_events.h"
+#include "kvm_private.h"
+
+/*
+ * Helpers
+ */
+static void
+fill_ev_common_kvmi_to_libvmi(
+    struct kvmi_dom_event *kvmi_event,
+    vmi_event_t *libvmi_event)
+{
+    //      standard regs
+    libvmi_event->x86_regs->rax = kvmi_event->event.common.arch.regs.rax;
```

Comment:
Would be faster to have a local pointer to `x86_regs` declared so you don't do double dereferences.